contracts/QuestFactory.sol

```diff
@@ -215,6 +237,93 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         return newQuest;
     }
 
+    /// @dev Create a NFT quest and start it at the same time. The function will transfer the total questFee amount to the QuestNFT
+    /// @param endTime_ The end time of the quest
+    /// @param startTime_ The start time of the quest
+    /// @param totalParticipants_ The total amount of participants (accounts) the quest will have
+    /// @param questId_ The id of the quest
+    /// @param jsonSpecCID_ The CID of the JSON spec for the quest
+    /// @param name_ The name of the quest
+    /// @param symbol_ The symbol of the quest
+    /// @param description_ The description of the quest
+    /// @param imageIPFSHash_ The IPFS hash of the image for the quest
+    /// @return address the quest contract address
+    function createQuestNFT(
+        uint256 endTime_,
+        uint256 startTime_,
+        uint256 totalParticipants_,
+        string memory questId_,
+        string memory jsonSpecCID_,
+        string memory name_,
+        string memory symbol_,
+        string memory description_,
+        string memory imageIPFSHash_
+    ) external payable returns (address) {
+        QuestNFTData memory data = QuestNFTData({
+            endTime: endTime_,
+            startTime: startTime_,
+            totalParticipants: totalParticipants_,
+            questId: questId_,
+            jsonSpecCID: jsonSpecCID_,
+            name: name_,
+            symbol: symbol_,
+            description: description_,
+            imageIPFSHash: imageIPFSHash_
+        });
+
+        address payable newQuest = createQuestNFTInternal(data);
+
+        QuestNFTContract(newQuest).initialize(
+            data.endTime,
+            data.startTime,
+            data.totalParticipants,
+            data.questId,
+            nftQuestFee,
+            protocolFeeRecipient,
+            address(this),
+            data.jsonSpecCID,
+            data.name,
+            data.symbol,
+            data.description,
+            data.imageIPFSHash
+        );
+
+        (bool success, ) = payable(newQuest).call{value: msg.value}("");
```

Comment:
What is the purpose of having this here? Could it also be called if `msg.value` is set to 0?

Reply:
Ahh requires `msg.value` >= `totalQuestNFTFee` when calling `createQuestNFTInternal()`